tests/test_app/main.go

```diff
@@ -27,8 +30,17 @@ func testCounter() {
 	}
 
 	fmt.Printf("Running %s test with abci=%s\n", abciApp, abciType)
-	appProc := startApp(abciApp)
-	defer appProc.StopProcess(true)
+	cmd := exec.Command("bash", "-c", fmt.Sprintf("abci-cli %s", abciApp))
+	cmd.Stdout = os.Stdout
+	if err := cmd.Start(); err != nil {
+		log.Fatalf("starting %q err: %v", abciApp, err)
+	}
+	defer cmd.Wait()
+	defer cmd.Process.Kill()
+
+	// TODO: A better to wait for the app to start?
```

Comment:
yes, we should be able to use `abci-cli echo` repeatedly until it works!

Reply:
Would you like me to incorporate this change in right now? That sleep was a port over from the original code :)

Reply:
Yeh I noticed. But I also fixed a bunch of these recently and non-deterministic tests are the worst, so would love if you fixed this!

Reply:
Awesome, coming up.

Reply:
C'est fini @ebuchman, merci beaucoup!